topydo/ui/Main.py

```diff
@@ -190,17 +197,47 @@ def _execute_handler(self, p_command, p_output=None):
             # TODO: show error message
             pass
 
-    def _update_all_columns(self):
+    def _update_all_columns(self, p_reset_keystate=False):
         for column, _ in self.columns.contents:
             column.update()
+            if p_reset_keystate:
+                column.keystate = None
 
     def _post_execute(self):
         # store dirty flag because base _post_execute will reset it after flush
         dirty = self.todolist.dirty
         super()._post_execute()
 
-        if dirty:
-            self._update_all_columns()
+        if dirty or self.pending_todos:
+            self._reset_state()
+
+    def _save_cmd(self, p_cmd, p_execute_signal):
+        verbosity = False
+        if p_execute_signal == 'execute_command':
+            verbosity = True
+        self._last_cmd = (p_cmd, verbosity)
+
+    def _repeat_last_cmd(self, p_todo_id=None):
+        try:
+            cmd, verbosity = self._last_cmd
+        except TypeError:
+            return
+
+        if verbosity:
+            output = self._output
+        else:
+            output = lambda _: None
+
+        if '{}' in cmd:
+            if not p_todo_id:
```

Comment:
Please add some comments here that you handle multiple selection here. Reading this code was not immediately obvious to me.